packages/editor/src/components/post-locked-modal/index.js

```diff
@@ -54,9 +67,61 @@ export default function PostLockedModal() {
 			activePostLock: getActivePostLock(),
 			postType: getPostType( getEditedPostAttribute( 'type' ) ),
 			previewLink: getEditedPostPreviewLink(),
+			isCurrentUserSessionActive:
+				__experimentalIsCurrentUserSessionActive(),
 		};
 	}, [] );
 
+	useEffect( () => {
+		const checkUserLoginHookName = `core/editor/check-user-login/post-locked-modal-${ instanceId }`;
+
+		/**
+		 * Binds to the Heartbeat Tick event and checks if the user is logged in or not.
+		 *
+		 * @param {Object} data Data received in the heartbeat request
+		 */
+		function checkUserLogin( data ) {
+			// Bail early, if the heartbeat has neither auth check data attribute nor nonce expiry attribute.
+			if (
+				! [ 'wp-auth-check', 'nonces_expired' ].some( ( attr ) =>
+					Object.prototype.hasOwnProperty.call( data, attr )
+				)
+			) {
+				return;
+			}
```

Comment:
`auth-check.js` in core only looks at `wp-auth-check`, not `nonces_expired`, see https://github.com/WordPress/wordpress-develop/blob/7b24083859c37d8ac75b9549f458bc389610eaef/src/js/_enqueues/lib/auth-check.js#L162-L168

So I'd remove the `nonces_expired` check here.

And then simplify to:

```suggestion
			if ( ! 'wp-auth-check' in data ) {
				return;
			}
```

Reply:
If we used `apiFetch` for the heartbeat request, then we wouldn't need to care about nonces at all, because `apiFetch` handles them internally (nonce middleware). Automatically requesting a new one when the old one expires etc.

I'm not sure if `admin-ajax.php` is compatible with `apiFetch` or whether we'd need to create a REST endpoint first.

Reply:
No that would require a REST endpoint. See the ither comments about that.

This comment here is still relevant for admin-ajax